app/viewmodels/regulatorsAndDocuments/GoverningDocumentStatusHelper.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.regulatorsAndDocuments
+
+import models.UserAnswers
+import models.regulators.SelectGoverningDocument
+import models.regulators.SelectGoverningDocument.Other
+import pages.QuestionPage
+import pages.regulatorsAndDocuments._
+import viewmodels.StatusHelper
+
+class GoverningDocumentStatusHelper extends StatusHelper {
+
+
+  private val allPages: Seq[QuestionPage[_]] =  Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage,
+    WhenGoverningDocumentApprovedPage, HasCharityChangedPartsOfGoverningDocumentPage, SectionsChangedGoverningDocumentPage)
+
+  private val validJourneys: Map[String, Seq[QuestionPage[_]]] = Map(
+    ("1", Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage)),
+    ("2", Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage,
+      WhenGoverningDocumentApprovedPage, HasCharityChangedPartsOfGoverningDocumentPage)),
+    ("3", Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage,
+      WhenGoverningDocumentApprovedPage, HasCharityChangedPartsOfGoverningDocumentPage, SectionsChangedGoverningDocumentPage)),
+    ("1A", Seq(SelectGoverningDocumentPage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage)),
+    ("2A", Seq(SelectGoverningDocumentPage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage,
+      HasCharityChangedPartsOfGoverningDocumentPage)),
+    ("3A", Seq(SelectGoverningDocumentPage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage,
+      HasCharityChangedPartsOfGoverningDocumentPage, SectionsChangedGoverningDocumentPage))
+  )
```

Comment:
this is a lot of repeated code and it's a bit of hack - the alternative is using a listbuffer or mutable variable, so it's fine, but it still makes me feel bad

Reply:
one thought however - this kind of approach doesn't seem very scalable for the larger sections - if you're happy with having different approaches in different sections, this is fine; otherwise we might consider having the mutable approach here, too